File: app/src/main/java/dev/ragnarok/fenrir/util/Side.java
package dev.ragnarok.fenrir.util;

import androidx.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@IntDef({Side.DISABLED, Side.NO_LOADING, Side.LOADING})
@Retention(RetentionPolicy.SOURCE)
public @interface Side {
    int DISABLED = 1;
    int NO_LOADING = 2;
    int LOADING = 3;
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/util/Side.kt
package dev.ragnarok.fenrir.util

import androidx.annotation.IntDef

@IntDef(Side.DISABLED, Side.NO_LOADING, Side.LOADING)
@Retention(AnnotationRetention.SOURCE)
annotation class Side {
    companion object {
        const val DISABLED = 1
        const val NO_LOADING = 2
        const val LOADING = 3
    }
}